packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -54,6 +58,82 @@ List<String> debugDescribeTransform(Matrix4 transform) {
   return matrix;
 }
 
+bool _extensionsInitialized = false;
+
+void initServiceExtensions() {
+  if (_extensionsInitialized)
+    return;
+
+  _extensionsInitialized = true;
+
+  assert(() {
+    registerExtension('flutter', _flutter);
+    // TODO: Expose debugDumpApp().
+    // TODO: Expose StatisticsOverlay.
+    registerExtension('flutter.debugPaint', _debugPaint);
+    registerExtension('flutter.timeDilation', _timeDilation);
+
+    // Emit an info level log message; this tells the debugger that the Flutter
+    // service extensions are registered.
+    log('Flutter initialized', name: 'flutter', level: 800);
+
+    return true;
+  });
+}
+
+/// Just respond to the request. Clients can use the existence of this call to
+/// know that the debug client is a Flutter app.
+Future<ServiceExtensionResponse> _flutter(String method, Map<String, dynamic> parameters) {
+  String result = JSON.encode({ 'type': '_extensionType', 'method': method });
+  return new Future<ServiceExtensionResponse>.value(new ServiceExtensionResponse.result(result));
+}
+
+/// Toggle the [debugPaintSizeEnabled] setting.
+Future<ServiceExtensionResponse> _debugPaint(String method, Map<String, dynamic> parameters) {
+  if (parameters.containsKey('enabled')) {
+    // TODO: This is a work around for a VM bug: sdk/25208 - all params are coerced to strings.
+    debugPaintSizeEnabled = parameters['enabled'].toString() == 'true';
+
+    // Redraw everything - mark the world as dirty.
+    RenderObjectVisitor visitor;
+    visitor = (RenderObject child) {
+      child.markNeedsPaint();
+      child.visitChildren(visitor);
+    };
+    Renderer.instance.renderView.visitChildren(visitor);
```

Comment:
`Renderer.instance?.renderView?.visitChildren`, since the debugger can presumably call into this code really, really early in the system's lifetime.